docs/querying_data.md

```diff
@@ -7,57 +7,131 @@ toc: false
+|sourceTable| Source Table Name. Needed to set hive environment properties. |  |
+|targetTable| Target Table Name. Needed for the intermediate storage directory structure.  |  |
+|sourceDataPath| Source DFS Base Path. This is where the Hudi metadata will be read. |  |
+|targetDataPath| Target DFS Base path. This is needed to compute the fromCommitTime. This is not needed if fromCommitTime is specified explicitly. |  |
+|tmpdb| The database to which the intermediate temp delta table will be created | hoodie_temp |
+|fromCommitTime| This is the most important parameter. This is the point in time from which the changed records are pulled from.  |  |
+|maxCommits| Number of commits to include in the pull. Setting this to -1 will include all the commits from fromCommitTime. Setting this to a value > 0, will include records that ONLY changed in the specified number of commits after fromCommitTime. This may be needed if you need to catch up say 2 commits at a time. | 3 |
+|help| Utility Help |  |
+
+
+Setting the fromCommitTime=0 and maxCommits=-1 will pull in the entire source dataset and can be used to initiate backfills. If the target dataset is a Hudi dataset,
+then the utility can determine if the target dataset has no commits or is behind more than 24 hour (this is configurable),
+it will automatically use the backfill configuration, since applying the last 24 hours incrementally could take more time than doing a backfill. The current limitation of the tool
+is the lack of support for self-joining the same table in mixed mode (normal and incremental modes).
 
 ## Spark
 
-There are two ways of running Spark SQL on Hudi datasets.
+Spark provides much easier deployment & management of Hudi jars and bundles into jobs/notebooks. At a high level, there are two ways to access Hudi datasets in Spark.
 
-First method involves, setting `spark.sql.hive.convertMetastoreParquet=false`, forcing Spark to fallback
-to using the Hive Serde to read the data (planning/executions is still Spark). This turns off optimizations in Spark
-towards Parquet reading, which we will address in the next method based on path filters.
-However benchmarks have not revealed any real performance degradation with Hudi & SparkSQL, compared to native support.
+ - **Hudi DataSource** : Supports Read Optimized, Incremental Pulls similar to how standard datasources (e.g: `spark.read.parquet`) work.
+ - **Read as Hive tables** : Supports all three views, including the real time view, relying on the custom Hudi input formats again like Hive.
+ 
+ In general, your spark job needs a dependency to `hoodie-spark` or `hoodie-spark-bundle-x.y.z.jar` needs to be on the class path of driver & executors (hint: use `--jars` argument)
+ 
+### Read Optimized table {#spark-ro-view}
 
-Sample command is provided below to spin up Spark Shell
+To read RO table as a Hive table using SparkSQL, all that needs to happen is to push a path filter into sparkContext as follows. 
```

Comment:
To read RO table as a Hive table using SparkSQL, simply push a path filter into sparkContext as follows :